pkg/common/telemetry/config.go

```diff
@@ -12,10 +12,11 @@ type MetricsConfig struct {
 }
 
 type FileConfig struct {
-	Prometheus *PrometheusConfig `hcl:"Prometheus"`
-	DogStatsd  []DogStatsdConfig `hcl:"DogStatsd"`
-	Statsd     []StatsdConfig    `hcl:"Statsd"`
-	M3         []M3Config        `hcl:"M3"`
+	Prometheus   *PrometheusConfig `hcl:"Prometheus"`
+	DogStatsd    []DogStatsdConfig `hcl:"DogStatsd"`
+	Statsd       []StatsdConfig    `hcl:"Statsd"`
+	M3           []M3Config        `hcl:"M3"`
+	DisableInMem bool              `hcl:"DisableInMem"`
```

Comment:
In case there is more configuration to be added to the "inmem" metrics later (however unlikely that is), we may want to play it safe and create an `InMemConfig` struct and include the disabled flag there. This also gives the config symmetry across the plugins.

Reply:
that could be good, but also then may be confusing that you still get an InMem sink even without declaring the InMemConfig in the telemetry config block (since we likely want to maintain compatibility for current users)

Reply:
I think I'm with @APTy on this one. Considering that in either case the inmem sink is treated differently than the rest (since it is on by default), I'd rather keep the config properly grouped. Stuff at the "global" level implies global impact on each sink type, which isn't the case for this configurable.

Also, i think named sections are `CamelCase` but individual configurables are `lower_case_with_underscores` (i.e. `disable_in_mem`, or `disabled` if it gets its own section)